events/ri/src/main/java/org/projectnessie/events/ri/kafka/KafkaEventSubscriber.java

```diff
@@ -0,0 +1,379 @@
+    }
+  }
+
+  /** Headers that are added to the Kafka records. */
+  public enum Header {
+    SPEC_VERSION("spec-version"),
+    REPOSITORY_ID("repository-id"),
+    INITIATOR("initiator"),
+    EVENT_TYPE("event-type"),
+    EVENT_CREATION_TIME("event-creation-time"),
+    ;
+
+    private final String key;
+
+    Header(String key) {
+      this.key = key;
+    }
+
+    public String getKey() {
+      return key;
+    }
+
+    public void addValue(Headers headers, byte[] value) {
+      headers.add(key, value);
+    }
+  }
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(KafkaEventSubscriber.class);
+  private static final Pattern COMMA = Pattern.compile(",");
+
+  // Accept all event types except: MERGE and TRANSPLANT
+  private static final EventTypeFilter EVENT_TYPE_FILTER =
+      EventTypeFilter.of(
+          EventType.COMMIT,
+          EventType.REFERENCE_CREATED,
+          EventType.REFERENCE_DELETED,
+          EventType.REFERENCE_UPDATED,
+          EventType.CONTENT_STORED,
+          EventType.CONTENT_REMOVED);
+
+  private final Properties props;
+  private final Function<Properties, Producer<String, Object>> producerFactory;
+
+  private byte[] serverSpecVersion;
+  private EventFilter eventFilter;
+  private String topicName;
+  private Producer<String, Object> producer;
+
+  /**
+   * Creates a new {@link KafkaEventSubscriber} using the configuration file specified by the system
+   * property {@value #NESSIE_KAFKA_PROPERTIES_FILE_SYS_PROP} or the environment variable {@value
+   * #NESSIE_KAFKA_PROPERTIES_FILE_ENV_VAR}.
+   *
+   * <p>Note: This default no-arg constructor MUST be present since that's the constructor used by
+   * {@link java.util.ServiceLoader}.
+   *
+   * @throws IOException if the configuration file cannot be read.
+   */
+  @SuppressWarnings("unused") // loaded via ServiceLoader
+  public KafkaEventSubscriber() throws IOException {
+    this(
+        System.getProperty(
+            NESSIE_KAFKA_PROPERTIES_FILE_SYS_PROP,
+            System.getenv(NESSIE_KAFKA_PROPERTIES_FILE_ENV_VAR) == null
+                ? NESSIE_KAFKA_PROPERTIES_FILE_DEFAULT
+                : System.getenv(NESSIE_KAFKA_PROPERTIES_FILE_ENV_VAR)));
+  }
+
+  public KafkaEventSubscriber(String location) throws IOException {
+    this(loadProperties(location));
+  }
+
+  public KafkaEventSubscriber(Properties props) {
+    this(props, KafkaProducer::new);
+  }
+
+  /**
+   * Creates a new {@link KafkaEventSubscriber} using the given {@link Properties} and {@link
+   * Producer} factory.
+   *
+   * <p>Intended for testing mostly.
+   */
+  public KafkaEventSubscriber(
+      Properties props, Function<Properties, Producer<String, Object>> producerFactory) {
+    this.props = props;
+    this.producerFactory = producerFactory;
+  }
+
+  @Override
+  public boolean isBlocking() {
+    // We don't wait for the broker acknowledgement, so we can tell Nessie
+    // that we don't do blocking I/O.
+    return false;
```

Comment:
how does the nessie handle the situation when `isBlocking==true`?

Reply:
It defers the call to `onXYZ` to a worker thread, as opposed to calling the method on the event loop thread.